### PR TITLE
Add unused but explanatory async

### PR DESCRIPTION
**Pull request by KnorpelSenf**
In response to https://github.com/grammyjs/grammY/issues/431

@dcdunkan are there any more such places? We have dozens or even hundreds of code examples with async handlers, but the rule is that anything that can be misunderstood will be misunderstood sooner or later. It's probably best to always add async, except for lambdas without block such as `(ctx) => ctx.reply("42")`.

### CODE REVIEW

File: site/docs/es/guide/middleware.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ Este stack de funciones es la _middleware stack_.
 (ctx, next) => ...    |
 ```
 
-Volviendo a nuestro ejemplo anterior, ahora sabemos por qué `bot.on(":photo")` ni siquiera se comprobó: el middleware en `bot.on(":text", (ctx) => { ... })` ya se encargó de la actualización, y no llamó a `next`.
```
Comment (dcdunkan) [inline site/docs/es/guide/middleware.md:87]: @KnorpelSenf I think here the async is unnecessary (and in the translations). What do you think?

File: site/docs/advanced/transformers.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ Calls via context objects of concurrently running middleware are left unaffected
 As soon as the respective middleware completes, the transformer function is discarded.
 
 ```ts
-bot.on("message", (ctx) => {
+bot.on("message", async (ctx) => {
```
Comment (KnorpelSenf) [inline site/docs/advanced/transformers.md:54]: We should call next here

File: site/docs/es/plugins/session.md [hunk 3]
```diff
@@ -669,7 +669,7 @@ En la práctica, en lugar de tener los datos de la sesión disponibles en `ctx.s
 
 ```ts
 // Sesiones por defecto (sesiones estrictas)
-bot.command("settings", (ctx) => {
+bot.command("settings", async (ctx) => {
```
Comment (KnorpelSenf) [inline site/docs/es/plugins/session.md:672]: Why is the highlighting broken here, is there no parse?

Comment (dcdunkan) [reply site/docs/es/plugins/session.md:672]: Not sure, but the file looks correct locally. Seems like just an highlighting issue on Github's side.

File: site/docs/advanced/transformers.md (hunk 2 above)
Comment (dcdunkan) [reply site/docs/advanced/transformers.md:54]: In this pull request, or should we do this in another?

File: site/docs/es/guide/middleware.md (hunk 1 above)
Comment (KnorpelSenf) [reply site/docs/es/guide/middleware.md:87]: I actually agree

File: site/docs/advanced/transformers.md (hunk 2 above)
Comment (KnorpelSenf) [reply site/docs/advanced/transformers.md:54]: In a subsequent PQ. It's kinda unrelated.

**Review by roj1512 — APPROVED** [review]
I am also approving these changes on behalf of Dunkan since we are in each other’s souls.

**Comment by KnorpelSenf** [comment]
Ready to merge then :)

**Comment by roj1512** [comment]
I will take that as it is not translatable.

**Pull request merged**